components/provider_cldr/src/transform/timezones/mod.rs

```diff
@@ -0,0 +1,366 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::cldr_langid::CldrLangID;
+use crate::error::Error;
+use crate::reader::{get_subdirectories, open_reader};
+use crate::CldrPaths;
+use icu_datetime::provider::{
+    key,
+    timezones::{
+        ExemplarCitiesV1, MetaZoneGenericNamesLongV1, MetaZoneGenericNamesShortV1,
+        MetaZoneSpecificNamesLongV1, MetaZoneSpecificNamesShortV1, TimeZoneFormatsV1,
+    },
+};
+use icu_provider::{
+    erased::ErasedDataProvider,
+    iter::{IterableDataProviderCore, KeyedDataProvider},
+    prelude::*,
+    serde::SerdeSeDataStruct,
+};
+use std::borrow::Cow;
+use std::convert::TryFrom;
+use std::marker::PhantomData;
+
+mod cldr_json;
+
+/// All keys that this module is able to produce.
+pub const ALL_KEYS: [ResourceKey; 6] = [
+    key::TIMEZONE_FORMATS_V1,
+    key::TIMEZONE_EXEMPLAR_CITIES_V1,
+    key::TIMEZONE_GENERIC_NAMES_LONG_V1,
+    key::TIMEZONE_GENERIC_NAMES_SHORT_V1,
+    key::TIMEZONE_SPECIFIC_NAMES_LONG_V1,
+    key::TIMEZONE_SPECIFIC_NAMES_SHORT_V1,
+];
+
+/// A data provider reading from CLDR JSON zones files.
+#[derive(PartialEq, Debug)]
+pub struct TimeZonesProvider<'d> {
+    data: Vec<(CldrLangID, cldr_json::LangTimeZones)>,
+    _phantom: PhantomData<&'d ()>, // placeholder for when we need the lifetime param
```

Comment:
`question`: my understanding was that `_` prefix is to be used for arguments and variables that are never assigned, but this is assigned in the constructor. Due to the nature of `phantom` I'd expect to see `_phantom` if it was the pattern, but I rather see it without `_` like in https://doc.rust-lang.org/std/marker/struct.PhantomData.html.
Is there a reason you diverged from the example?

Reply:
I chose this format to be congruent with the pre-existing dates provider: 

https://github.com/unicode-org/icu4x/blob/5f723c3ece94dec75f0c39166f34ff51c8d896b0/components/provider_cldr/src/transform/dates.rs#L21-L26

But I agree with you. I think it would be best to match the rust-lang examples.